lms/djangoapps/instructor/tests/test_certificates.py

```diff
@@ -458,35 +453,255 @@ def test_certificate_exception_same_user_in_two_different_courses(self):
+        )
+        # Assert error on request
+        self.assertEqual(response.status_code, 400)
+
+        res_json = json.loads(response.content)
+
+        # Assert Request not successful
+        self.assertFalse(res_json['success'])
+        # Assert Error Message
+        self.assertEqual(
+            res_json['message'],
+            u"Invalid request, Please refresh the page and try again."
+        )
+
+        # Verify that certificate exception is not removed in case of error in request
+        certificate_exception = CertificateWhitelist.objects.get(user=self.user, course_id=self.course.id)
+        self.assertEqual(certificate_exception.user, self.user)
+        self.assertEqual(certificate_exception.course_id, self.course.id)
+        self.assertTrue(certificate_exception.whitelist)
+
+    def test_remove_certificate_exception_non_existing_error(self):
+        """
+        Test certificates exception removal api endpoint returns error
+        when called with non existing certificate exception id
+        """
+        invalid_certificate_exception_id = 0  # pylint: disable=invalid-name
+        CertificateWhitelistFactory.create(
+            user=self.user,
+            course_id=self.course.id,
+            whitelist=True,
+            notes='Test Notes'
+        )
+
+        self.client.login(username=self.global_staff.username, password='test')
+        url = reverse(
+            'certificate_exception_view',
+            kwargs={'course_id': unicode(self.course.id), 'certificate_exception_id': invalid_certificate_exception_id}
+        )
+
+        response = self.client.delete(
+            url,
             content_type='application/json'
         )
+        # Assert error on request
+        self.assertEqual(response.status_code, 400)
+
+        res_json = json.loads(response.content)
+
+        # Assert Request not successful
+        self.assertFalse(res_json['success'])
+        # Assert Error Message
+        self.assertEqual(
+            res_json['message'],
+            u"Certificate exception [id={}] does not exist in "
+            u"certificate white list.".format(invalid_certificate_exception_id)
+        )
+
+        # Verify that valid certificate exception is not removed in case of error in request
+        certificate_exception = CertificateWhitelist.objects.get(user=self.user, course_id=self.course.id)
+        self.assertEqual(certificate_exception.user, self.user)
+        self.assertEqual(certificate_exception.course_id, self.course.id)
+        self.assertTrue(certificate_exception.whitelist)
+
+    def test_generate_certificate_exceptions_all_students(self):
+        """
+        Test generate certificates exceptions api endpoint returns success
+        when called with existing certificate exception
+        """
+        certificate_exception = CertificateWhitelistFactory.create(
+            user=self.user,
+            course_id=self.course.id,
+            whitelist=True,
```

Comment:
Can we put default values of `whitelist` and `notes` in `CertificateWhitelistFactory` instead of putting them in tests?